scripts/generate_schema_docs/main.go

```diff
@@ -0,0 +1,140 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Generates configs from the full example, inits all modules, and checks if any resources are not supported by the importer.
+// Meant to be run from the repo root like so:
+// go run ./scripts/generate_schema_docs
+
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/healthcare-data-protection-suite/internal/hcl"
+)
+
+var (
+	recipesDir = flag.String("recipes_dir", "templates/tfengine/recipes", "Directory hosting Terraform engine recipes.")
+	outputDir  = flag.String("output_dir", "docs/tfengine/recipes", "Directory to output markdown files.")
+)
+
+var schemaRE = regexp.MustCompile(`(?s)(?:schema = {)(.+?)(?:}\n\n)`)
+
+func main() {
+	if err := run(*recipesDir, *outputDir); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func run(recipesDir, outputDir string) error {
+	tmp, err := ioutil.TempDir("", "")
+	if err != nil {
+		return err
+	}
+	defer os.RemoveAll(tmp)
+
+	fn := func(path string, info os.FileInfo, err error) error {
```

Comment:
This function is pretty big. Consider extracting parts of it out (ex. `func matches(path string) [][]byte`).

Reply:
Hmm, since we need to return nil in the case there are no matches this may be cleaner to just leave as is. Do you have a better alternative?

Reply:
I think you can extract several functions, I tried at https://play.golang.org/p/oYUXpCsd5UB

Reply:
findMatches is not implemented quite correctly. We would need to check for len 0 twice (once in findMatches then again in the walk func). I'll do the schemaFromHCL cleanup.

Reply:
Why do you need to check for length 0 twice?

Reply:
Oh, because I guess it's not an error to have 0 matches. Still, I think separating out these functions makes it more readable.

Reply:
Done